src/storage/remote/protocol.c

```diff
@@ -158,69 +106,214 @@ storageRemoteFeatureProtocol(const VariantList *paramList, ProtocolServer *serve
         }
 
         // Return storage features
-        protocolServerWriteLine(server, jsonFromStr(storagePathP(storage, NULL)));
-        protocolServerWriteLine(server, jsonFromUInt64(storageInterface(storage).feature));
+        PackWrite *result = protocolPackNew();
+        pckWriteStrP(result, storagePathP(storage, NULL));
+        pckWriteU64P(result, storageInterface(storage).feature);
 
-        protocolServerResponse(server, NULL);
+        protocolServerDataPut(server, result);
+        protocolServerDataEndPut(server);
     }
     MEM_CONTEXT_TEMP_END();
 
     FUNCTION_TEST_RETURN_VOID();
 }
 
 /**********************************************************************************************************************************/
+typedef struct StorageRemoteInfoProcotolWriteData
+{
+    MemContext *memContext;                                         // Mem context used to store values from last call
+    time_t timeModifiedLast;                                        // timeModified from last call
+    mode_t modeLast;                                                // mode from last call
+    uid_t userIdLast;                                               // userId from last call
+    gid_t groupIdLast;                                              // groupId from last call
+    String *user;                                                   // user from last call
+    String *group;                                                  // group from last call
+} StorageRemoteInfoProtocolWriteData;
+
+// Helper to write storage info into the protocol. This function is not called unless the info exists so no need to write exists or
+// check for level == storageInfoLevelExists.
+//
+// Fields that do not change from one call to the next are omitted to save bandwidth.
+static void
+storageRemoteInfoProtocolPut(
+    StorageRemoteInfoProtocolWriteData *const data, PackWrite *const write, const StorageInfo *const info)
+{
+    FUNCTION_TEST_BEGIN();
+        FUNCTION_TEST_PARAM_P(VOID, data);
+        FUNCTION_TEST_PARAM(STORAGE_INFO, info);
+    FUNCTION_TEST_END();
+
+    ASSERT(data != NULL);
+    ASSERT(info != NULL);
+
+    // Write type and time
+    pckWriteU32P(write, info->type);
+    pckWriteTimeP(write, info->timeModified - data->timeModifiedLast);
```

Comment:
Name is only written for `InfoList` because for `Info` the caller already knows the filename because it was in the request. The caller also knows the level, so `or check for level == storageInfoLevelExists` means there is no logic based on  `storageInfoLevelExists`. Exists only needs to be sent for `Info` (line 250) but does not need to be written to `infoList` since all files exist by definition (since they were just stat'd).

I suppose we could just write all the fields all the time, but that's not what we did before so I didn't change it. Bytes count...